### DISCUSSION TITLE
More image editor tools

### DISCUSSION THREAD

**Discussion by ikcii** [hlky/stable-diffusion-webui#675] [Feature suggestions]
It would be amazing to have more tools for editing images to make it easier to re-generate just specific parts of the image, such as:
- pixelate brush instead of only pixelate selected area with changeable brush size and pixels density,
- random noise brush,
- uh that's mostly everything I could come up with but also:
I think it'd be quite cool if there was an option to make the UI automatically see which pixels are changed when you compare initial image vs edited image and then use that difference as a mask to make the AI regenerate only the areas that got changed while editing

**Comment by beenotung** [comment]
It would be great a drawing canvas is added on top of the input image similar to [1] [2], this can enable the user to refine the concept in the image within the web UI.

Right now the user can download the image, then edit with other tools e.g. GIMP, then upload back as input image, so this feature suggestion may have relative low priority but it would be an improvement on UX

[1] https://huggingface.co/spaces/huggingface/diffuse-the-rest
[2] https://www.drawanything.app/
